### PR TITLE
Make test_daemon.py robust to discovery latency.

### PR DESCRIPTION
**Pull request by hidmic**
Fixes https://github.com/ros2/ros2cli/issues/503. Some RMW implementations are slow enough to make this graph API-intensive test flakey. This pull requests attempts to circumvent that problem.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
I don't think we should add yet another layer of retry logic here. Why can't the module level fixture not have a sleep instead?

File: ros2cli/test/test_daemon.py [hunk 1]
```diff
@@ -110,6 +110,33 @@ def daemon_node():
     assert spawn_daemon(args=[], wait_until_spawned=5.0)
     with DaemonNode(args=[]) as node:
         yield node
```
Comment (jacobperron) [inline ros2cli/test/test_daemon.py:112]: Is it sufficient to wait for discovery of the test node at the fixture level, here? E.g. wait until `get_node_names_and_namespaces()` discovers the test node.

**Comment by hidmic** [comment]
> Why can't the module level fixture not have a sleep instead?

A single sleep is brittle.

> Is it sufficient to wait for discovery of the test node at the fixture level, here? E.g. wait until get_node_names_and_namespaces() discovers the test node.

That's something we can try, indeed.

File: ros2cli/test/test_daemon.py (hunk 1 above)
Comment (hidmic) [reply ros2cli/test/test_daemon.py:124]: See https://github.com/ros2/ros2cli/pull/504/commits/559de8eba42069fe46e3e40bf6af9c5e9b68d1b8.

File: ros2cli/test/test_daemon.py [hunk 2]
```diff
@@ -104,12 +104,25 @@ def noop_execute_callback(goal_handle):
 
 @pytest.fixture(scope='module')
 def daemon_node():
+    import time
```
Comment (dirk-thomas) [inline ros2cli/test/test_daemon.py:107]: Nitpick: why a local import?

Comment (hidmic) [reply ros2cli/test/test_daemon.py]: No particular reason, see https://github.com/ros2/ros2cli/pull/504/commits/9d7f5d34dc75bcdb01e2e05e49cd98f81f5ba2c0.

**Comment by hidmic** [comment]
CI up to `ros2cli`, `rclpy`, and `rmw_connext` (when possible):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10632/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6064/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8657/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10538/)

**Pull request merged**